jobs/vqsr.py

```diff
@@ -0,0 +1,1033 @@
+        # Importing dynamically to make sure $CPG_DATASET_GCP_PROJECT is set.
+        from analysis_runner import dataproc
+
+        assert meta_ht_path
+        assert hard_filter_ht_path
+        job_name = 'VQSR: MT to site-only VCF'
+        combined_vcf_path = tmp_prefix / 'input.vcf.gz'
+        if not can_reuse(combined_vcf_path, overwrite):
+            mt_to_vcf_job = dataproc.hail_dataproc_job(
+                b,
+                f'jobs/dataproc_scripts/mt_to_siteonlyvcf.py --overwrite '
+                f'--mt {input_vcf_or_mt_path} '
+                f'--meta-ht {meta_ht_path} '
+                f'--hard-filtered-samples-ht {hard_filter_ht_path} '
+                f'-o {combined_vcf_path} ',
+                max_age='8h',
+                packages=[
+                    'cpg_utils',
+                    'cpg_gnomad',  # github.com/populationgenomics/gnomad_methods
+                    'click',
+                    'google',
+                    'fsspec',
+                    'sklearn',
+                    'gcloud',
+                    'selenium',
+                ],
+                num_secondary_workers=scatter_count,
+                # hl.export_vcf() uses non-preemptible workers' disk to merge VCF files.
+                # 10 samples take 2.3G, 400 samples take 60G, which roughly matches
+                # `huge_disk` (also used in the AS-VQSR VCF-gather job)
+                worker_boot_disk_size=huge_disk,
+                job_name=job_name,
+            )
+        else:
+            mt_to_vcf_job = b.new_job(f'{job_name} [reuse]', job_attrs)
+        jobs.append(mt_to_vcf_job)
+        tabix_job = add_tabix_job(
+            b,
+            vcf_path=combined_vcf_path,
+            disk_size=medium_disk,
+            job_attrs=job_attrs,
+        )
+        tabix_job.depends_on(mt_to_vcf_job)
+        assert isinstance(tabix_job.output_vcf, hb.ResourceGroup)
+        siteonly_vcf = tabix_job.output_vcf
+    else:
+        siteonly_vcf = b.read_input_group(
+            **{
+                'vcf.gz': str(input_vcf_or_mt_path),
+                'vcf.gz.tbi': str(input_vcf_or_mt_path) + '.tbi',
+            }
+        )
+
+    indel_recalibrator_j = indel_recalibrator_job(
+        b=b,
+        sites_only_variant_filtered_vcf=siteonly_vcf,
+        mills_resource_vcf=resources['mills'],
+        axiom_poly_resource_vcf=resources['axiom_poly'],
+        dbsnp_resource_vcf=resources['dbsnp'],
+        disk_size=small_disk,
+        use_as_annotations=use_as_annotations,
+        is_small_callset=is_small_callset,
+        job_attrs=job_attrs,
+    )
+    # To make type checkers happy:
+    assert isinstance(indel_recalibrator_j.recalibration, hb.ResourceGroup)
+    assert isinstance(indel_recalibrator_j.tranches, hb.ResourceFile)
+    jobs.append(indel_recalibrator_j)
+
+    snp_max_gaussians = 6
+    if is_small_callset:
+        snp_max_gaussians = 4
+    elif is_huge_callset:
+        snp_max_gaussians = 8
```

Comment:
What's the rationale for changing these depending on the size of the callset?

Reply:
That was what they used in WARP before, but now it's just 6 consistently for SNPs, changed that accordingly